code/modules/mob/living/simple_animal/bot/mulebot.dm

```diff
@@ -706,11 +706,15 @@
 	H.apply_damage(0.5*damage, BRUTE, "r_arm", run_armor_check("r_arm", "melee"))
 
 
-	var/turf/T = get_turf(src)
-	H.add_mob_blood(H)
-	H.add_splatter_floor(T)
 
-	bloodiness += 4
+
+	if(NO_BLOOD in H.dna.species.species_traits)//Does the run over mob have blood?
+		return//If it doesn't it shouldn't bleed (Though a check should be made eventually for things with liquid in them, like slime people, vox armalis, etc.)
+	else
```

Comment:
Good work.

Improvement on your code though, you don't actually need the else.

As you return on the IF above it, the code below will only ever run if it fails anyway.